File: Examples/test-suite/ocaml/char_binary_runme.ml
open Swig
open Char_binary

let _ =
  let t = new_Test '() in
  assert (t -> strlen ("hile") as int = 4);
  assert (t -> ustrlen ("hile") as int = 4);
  assert (t -> strlen ("hil\x00") as int = 4);
  assert (t -> ustrlen ("hil\x00") as int = 4);

  let pc = _new_pchar '(5) in
  assert (_pchar_setitem '(pc, 0, 'h') = C_void);
  assert (_pchar_setitem '(pc, 1, 'o') = C_void);
  assert (_pchar_setitem '(pc, 2, 'l') = C_void);
  assert (_pchar_setitem '(pc, 3, 'a') = C_void);
  assert (_pchar_setitem '(pc, 4, 0) = C_void);
  assert (t -> strlen (pc) as int = 4);
  assert (t -> ustrlen (pc) as int = 4);
  ignore (_var_pchar '(pc));
  assert (_var_pchar '() as string = "hola");
  ignore (_var_namet '(pc));
  assert (_var_namet '() as string = "hola");
  _delete_pchar(pc)
;;

File: Examples/test-suite/octave/char_binary_runme.m
char_binary

t = char_binary.Test();
str = "hile";
if (t.strlen(str) != 4)
  disp(t.strlen(str));
  error("bad multi-arg typemap");
endif
if (t.ustrlen(str) != 4)
  disp(t.ustrlen(str));
  error("bad multi-arg typemap");
endif

if (t.strlen("hil\000") != 4)
  error("bad multi-arg typemap");
endif
if (t.ustrlen("hil\000") != 4)
  error("bad multi-arg typemap");
endif

% creating a raw char*
pc = new_pchar(5);
pchar_setitem(pc, 0, 'h');
pchar_setitem(pc, 1, 'o');
pchar_setitem(pc, 2, 'l');
pchar_setitem(pc, 3, 'a');
pchar_setitem(pc, 4, 0);

if (0)
  % FIXME: Can not convert to const (SWIG_TypeError)
  if (t.strlen(pc) != 4)
    error("bad multi-arg typemap");
  endif

  % FIXME: Can not convert to const (SWIG_TypeError)
  if (t.ustrlen(pc) != 4)
    error("bad multi-arg typemap");
  endif
endif

char_binary.var_pchar = pc;
if (0)
  % FIXME: could not dispatch binary operator
  if (char_binary.var_pchar != "hola")
    error("bad pointer case");
  endif
endif

char_binary.var_namet = pc;
if (0)
  % FIXME: could not dispatch binary operator
  if (char_binary.var_namet != "hola")
    error("bad pointer case");
  endif
endif

delete_pchar(pc);
